pulsar-broker/src/main/java/org/apache/pulsar/broker/service/DrainingHashesTracker.java

```diff
@@ -34,14 +36,15 @@
 import org.roaringbitmap.RoaringBitmap;
 
 /**
- * A thread-safe map to store draining hashes in the consumer.
+ * A thread-safe map to store draining hashes in the consumer using read-write locks for improved concurrency.
```

Comment:
is this necessary to access this instance by multiple threads? Can we run this ops by a single thread?

Reply:
Yes, it's necessary. The draining hashes state is exposed in topic stats and multi-thread access for stats cannot be easily eliminated without additional complexity.